cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/JsonMenu/LiveViewer/index.tsx

```diff
@@ -0,0 +1,197 @@
+import ConfigurationGroup from 'components/ConfigurationGroup';
+import If from 'components/If';
+import LoadingSVGCentered from 'components/LoadingSVGCentered';
+import { JSONStatusMessage } from 'components/PluginJSONCreator/Create/Content/JsonMenu';
+import JsonActionButtons from 'components/PluginJSONCreator/Create/Content/JsonMenu/JsonActionButtons';
+import { ICreateContext } from 'components/PluginJSONCreator/CreateContextConnect';
+import * as React from 'react';
+
+export enum LiveViewMode {
+  JSONView = 'JSON_VIEW',
+  ConfigurationGroupsView = 'CONFIGURATION_GROUPS',
+}
+
+const LIVE_VIEWER_WIDTH = '600px';
+
+const styles = (theme): StyleRules => {
+  return {
+    liveViewer: {
+      zIndex: theme.zIndex.drawer,
+      width: LIVE_VIEWER_WIDTH,
+    },
+    liveViewerPaper: {
+      width: LIVE_VIEWER_WIDTH,
+      backgroundColor: theme.palette.white[50],
+    },
+    toolbar: {
+      minHeight: '48px',
+    },
+    mainMenu: {
+      width: '100%',
+      height: '100%',
+      borderTop: `1px solid ${theme.palette.grey['500']}`,
+      paddingTop: theme.Spacing(1),
+      paddingBottom: theme.Spacing(1),
+    },
+    jsonFailStatus: {
+      position: 'relative',
+      margin: '0 auto',
+      color: theme.palette.red[50],
+      fontFamily: 'Courier New',
+    },
+    jsonSuccessStatus: {
+      position: 'relative',
+      margin: '0 auto',
+      color: theme.palette.blue[50],
+      fontFamily: 'Courier New',
+    },
+    liveViewLoadBox: {
+      width: '100%',
+      height: '100%',
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+    },
+    jsonLiveCode: {
+      padding: '14px',
+    },
+  };
+};
+
+interface ILiveViewerProps extends WithStyles<typeof styles>, ICreateContext {
+  JSONConfig: any;
+  collapseLiveView: () => void;
+  onDownloadClick: () => void;
+  populateImportResults: (filename: string, fileContent: string) => void;
+  JSONFilename: string;
+  JSONStatus: JSONStatusMessage;
+  downloadDisabled: boolean;
+  JSONErrorMessage: string;
+}
+
+const LiveViewerView: React.FC<ILiveViewerProps> = ({
+  classes,
+  JSONConfig,
+  collapseLiveView,
+  onDownloadClick,
+  populateImportResults,
+  JSONFilename,
+  JSONStatus,
+  downloadDisabled,
+  JSONErrorMessage,
+}) => {
+  const [liveViewMode, setLiveViewMode] = React.useState(LiveViewMode.ConfigurationGroupsView);
+
+  // Values needed for Configuration Groups live view
+  const [pluginProperties, setPluginProperties] = React.useState(null);
+  const [values, setValues] = React.useState<Record<string, string>>({});
+
+  const [loading, setLoading] = React.useState(false);
+
+  // When JSON config changes, show loading view for 500ms
+  // This is in order to force rerender ConfigurationGroup component
+  React.useEffect(() => {
+    // after a setTimeout for 500ms, set the loading state back to false
+    setLoading(true);
+
+    // for rerendering ConfigurationGroup
```

Comment:
`debounce` is added at refactor PR #12310 https://github.com/cdapio/cdap/blob/882eb810d3291ed6b39efaf6394a932584abb867/cdap-ui/app/cdap/components/PluginJSONCreator/Create/PluginJSONMenu/LiveViewMenu/LiveConfigurationGroup/index.tsx